cluster/swarm/cluster.go

```diff
@@ -347,6 +347,60 @@ func (c *Cluster) Load(imageReader io.Reader, callback func(what, status string)
 	wg.Wait()
 }
 
+// Import image
+func (c *Cluster) Import(source string, repository string, tag string, imageReader io.Reader, callback func(what, status string)) {
+	var wg sync.WaitGroup
+	c.RLock()
+	pipeWriters := []*io.PipeWriter{}
+	pipeReaders := []*io.PipeReader{}
+	for _, n := range c.engines {
+		wg.Add(1)
+
+		pipeReader, pipeWriter := io.Pipe()
+		pipeReaders = append(pipeReaders, pipeReader)
+		pipeWriters = append(pipeWriters, pipeWriter)
+
+		go func(reader *io.PipeReader, nn *cluster.Engine) {
+			defer wg.Done()
+			defer reader.Close()
+
+			// call engine import
+			err := nn.Import(source, repository, tag, reader)
+			if callback != nil {
+				if err != nil {
+					callback(nn.Name, err.Error())
+				} else {
+					callback(nn.Name, "Import success")
+				}
+			}
+
+		}(pipeReader, n)
+	}
+
+	// create multi-writer
+	listWriter := []io.Writer{}
+	for _, pipeW := range pipeWriters {
+		listWriter = append(listWriter, pipeW)
+	}
+	mutiWriter := io.MultiWriter(listWriter...)
+
+	// copy image-reader to muti-writer
+	_, err := io.Copy(mutiWriter, imageReader)
+	if err != nil {
+		log.Error(err)
+	}
+
+	// close pipe writers
+	for _, pipeW := range pipeWriters {
+		pipeW.Close()
+	}
+
+	c.RUnlock()
```

Comment:
`RUnlock` must happen **immediately** after `for _, n := range c.engines {`.

Otherwise, this will lock the entire cluster down until the entire `io.Copy` is finished